### ISSUE TITLE
Nogo errors are not reported when recompiling external tests

### ISSUE THREAD

**Issue by linzhp** [bazelbuild/rules_go#2660]
### What version of rules_go are you using?

0.24.2

### What version of gazelle are you using?

0.22.0

### What version of Bazel are you using?

3.5

### Does this issue reproduce with the latest releases of all the above?

Yes

### What operating system and processor architecture are you using?

Darwin amd64

### What did you do?

```bash
$ git clone git@github.com:linzhp/bazel_examples.git
$ cd bazel_examples
$ git checkout c9c2b93
$ bazel build //recompile:all
```

### What did you expect to see?

```
ERROR: /Users/zplin/gocode/src/github.com/linzhp/bazel_examples/recompile/BUILD.bazel:10:8: GoCompilePkg recompile/go_default_test_test.external.a failed (Exit 1) builder failed: error executing command bazel-out/host/bin/external/go_sdk/builder compilepkg -sdk external/go_sdk -installsuffix darwin_amd64 -src recompile/foo_test.go -arc ... (remaining 21 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
compilepkg: nogo: errors found by nogo during build-time code analysis:
/private/var/tmp/_bazel_zplin/1b23f78923ad6c0028e8aab1a0a6d5fb/sandbox/darwin-sandbox/126/execroot/__main__/recompile/foo_test.go:9:9: github.com/linzhp/bazel_examples/recompile.Thing composite literal uses unkeyed fields
INFO: Elapsed time: 3.191s, Critical Path: 0.30s
INFO: 3 processes: 3 darwin-sandbox.
FAILED: Build did NOT complete successfully
```

This error can be seen if `foo_internal_test.go` is removed and the `//recompile:go_default_test` depends on `//recompile:go_default_library` instead of embedding it.

### What did you see instead?
```
INFO: Analyzed 2 targets (59 packages loaded, 9703 targets configured).
INFO: Found 2 targets...
INFO: Elapsed time: 22.320s, Critical Path: 5.97s
INFO: 31 processes: 31 darwin-sandbox.
INFO: Build completed successfully, 32 total actions
```

**Comment by jayconrod** [comment]
I was able to reproduce this, but it looks like this is the intended behavior of that analyzer: see [`isLocalType`](https://github.com/golang/tools/blob/f1e51e6b9437f17500263287e3b9c46707f3c8c7/go/analysis/passes/composite/composite.go#L114). There's a special case for external tests of the same package.

I tried changing the example to use `fmt.Printf("%s")` which triggers the `printf` analyzer, and nogo reported errors for both internal and external tests.
